src/main/java/org/privacyidea/PrivacyIDEA.java

```diff
@@ -229,17 +229,29 @@ public PIResponse triggerChallenges(String username, Map<String, String> headers
      * Poll for status of the given transaction ID once.
      *
      * @param transactionID transaction ID to poll for
-     * @return the status value, true or false
+     * @return the challenge status or "ChallengeStatus.none" if error
      */
-    public boolean pollTransaction(String transactionID)
+    public ChallengeStatus pollTransaction(String transactionID)
     {
         Objects.requireNonNull(transactionID, "TransactionID is required!");
 
         Map<String, String> params = new LinkedHashMap<>();
         params.put(TRANSACTION_ID, transactionID);
         String response = runRequestAsync(ENDPOINT_POLLTRANSACTION, params, Collections.emptyMap(), false, GET);
         PIResponse piresponse = this.parser.parsePIResponse(response);
-        return piresponse.value;
+        if (piresponse.challengeStatus == ChallengeStatus.pending)
```

Comment:
`piresponse.challengestatus` is of type `challengestatus` so why not just return that instead of all these if else

Reply:
[8d1a230](https://github.com/privacyidea/java-client/pull/68/commits/8d1a230e1d68fe141aaecf42e51da91d8ca4e1e9)